sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication
+{
+    /// <summary>
+    /// Represents a credential that exchanges an Entra token for an Azure Communication Services (ACS) token, enabling access to ACS resources.
+    /// </summary>
+    internal sealed class EntraTokenCredential : ICommunicationTokenCredential
+    {
+        private HttpPipeline _pipeline;
+        private string _resourceEndpoint;
+        private readonly ThreadSafeRefreshableAccessTokenCache _accessTokenCache;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EntraTokenCredential"/>.
+        /// </summary>
+        /// <param name="options">The options for how the token will be fetched</param>
+        /// <param name="pipelineTransport">Only for testing.</param>
+        public EntraTokenCredential(EntraCommunicationTokenCredentialOptions options, HttpPipelineTransport pipelineTransport = null)
+        {
+            this._resourceEndpoint = options.ResourceEndpoint;
+            _pipeline = CreatePipelineFromOptions(options, pipelineTransport);
+            _accessTokenCache = new ThreadSafeRefreshableAccessTokenCache(
+                    ExchangeEntraToken,
+                    ExchangeEntraTokenAsync,
+                    false, null, null);
+        }
```

Comment:
if we all agree to fetch immediately, we could call `GetTokenAsync` without awaiting it from within the constructor